kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java

```diff
@@ -187,10 +201,47 @@ private void setupTable() throws DBException {
       schema = new Schema(columns);
 
       CreateTableOptions builder = new CreateTableOptions();
-      builder.setRangePartitionColumns(new ArrayList<String>());
-      List<String> hashPartitionColumns = new ArrayList<>();
-      hashPartitionColumns.add(KEY);
-      builder.addHashPartitions(hashPartitionColumns, numTablets);
+
+      if (partitionSchema.equals("hashPartition")) {
+        builder.setRangePartitionColumns(new ArrayList<String>());
+        List<String> hashPartitionColumns = new ArrayList<>();
+        hashPartitionColumns.add(KEY);
+        builder.addHashPartitions(hashPartitionColumns, numTablets);
+      } else if (partitionSchema.equals("rangePartition")) {
+        if (prop.getProperty(CoreWorkload.INSERT_ORDER_PROPERTY, CoreWorkload.INSERT_ORDER_PROPERTY_DEFAULT)
+            .compareTo("ordered") != 0) {
+          throw new DBException("Must specify `insertorder=ordered` if using rangePartition schema.");
+        }
+
+        String maxKeyValue = String.valueOf(recordCount);
+        if (zeropadding < maxKeyValue.length()) {
+          throw new DBException(String.format("Invalid zeropadding value: %d, zeropadding needs to be larger "
+              + "or equal to number of digits in the record number: %d.", zeropadding, maxKeyValue.length()));
+        }
+
+        List<String> rangePartitionColumns = new ArrayList<>();
+        rangePartitionColumns.add(KEY);
+        builder.setRangePartitionColumns(rangePartitionColumns);
+        // Add rangePartitions
+        long lowerNum = 0;
+        long upperNum = 0;
+        int remainder = (int) recordCount % numTablets;
+        for (int i = 0; i < numTablets; i++) {
+          lowerNum = upperNum;
+          upperNum = lowerNum + recordCount / numTablets;
+          if (i < remainder) {
+            ++upperNum;
+          }
+          PartialRow lower = schema.newPartialRow();
```

Comment:
Should any secondary hash partitioning be used to prevent insert workloads targeting 1 tablet at a time? Perhaps optionally.

Reply:
`kudu_pre_split_num_tablets` defines the number of tablets, if we use both range and hash partitioning,  it will be hard to determine partition schema.